nixos/modules/config/ldap.nix

```diff
@@ -232,21 +256,31 @@ in
         wantedBy = [ "multi-user.target" ];
 
         preStart = ''
-          mkdir -p /run/nslcd
-          rm -f /run/nslcd/nslcd.pid;
-          chown nslcd.nslcd /run/nslcd
-          ${optionalString (cfg.bind.distinguishedName != "") ''
-            if test -s "${cfg.bind.password}" ; then
-              ln -sfT "${cfg.bind.password}" /run/nslcd/bindpw
-            fi
-          ''}
+          umask 0077
+          conf="$(mktemp)"
+          {
+            cat ${nslcdConfig.source}
+            test -z "${cfg.bind.distinguishedName}" -o ! -f "${cfg.bind.password}" ||
+            printf 'bindpw %s\n' "$(cat ${cfg.bind.password})"
+            test -z "${cfg.daemon.rootpwmoddn}" -o ! -f "${cfg.daemon.rootpwmodpw}" ||
+            printf 'rootpwmodpw %s\n' "$(cat ${cfg.daemon.rootpwmodpw})"
```

Comment:
I've pushed a change quoting `${cfg.bind.password}` and `${cfg.daemon.rootpwmodpw}` in the `cat` since here someone could be using (very unwise) paths with spaces. What kind of escaping do you have in mind?